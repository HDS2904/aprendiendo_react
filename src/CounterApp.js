import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ( { value } ) => {

    // Uso de Hooks
    const [ counter, setCounter ] = useState( 0 );

    //Llamado de la funcion obtenida al desestructurar useState
    const handleAdd = ( e ) => {
        setCounter(counter + 1);
        // setCounter( ( c ) => c + 1 )
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            <button onClick={ handleAdd }> +1 </button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp;