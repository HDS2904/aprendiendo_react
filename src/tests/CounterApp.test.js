import React from 'react'
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('Prueba del componente CounterApp', () => {
    test('El componente debe tener una estructura HTML correcta', () => {
        const wrapper = shallow( <CounterApp /> )
        expect( wrapper ).toMatchSnapshot()
    })
    test('Compoenete debe retornar la linea evaluada segun el valor enviado 100', () => {
        const lineEval = shallow( <CounterApp value={ 100 } /> )
        const line = lineEval.find('h2').text().trim()
        expect( line ).toBe( '100' )
    })
    
    
})
