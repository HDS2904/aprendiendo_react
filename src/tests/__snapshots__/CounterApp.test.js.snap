// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente CounterApp El componente debe tener una estructura HTML correcta 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
     
     
  </h2>
  <button
    onClick={[Function]}
  >
     +1 
  </button>
  <button
    onClick={[Function]}
  >
     reset 
  </button>
  <button
    onClick={[Function]}
  >
     -1 
  </button>
</Fragment>
`;
